eth/abc.py

```diff
@@ -631,10 +631,33 @@ def persist_block(self,
             as genesis. Providing a ``genesis_parent_hash`` allows storage of blocks that
             aren't (yet) connected back to the true genesis header.
 
-        Assumes all block transactions have been persisted already.
+        WARNING: This API assumes all block transactions have been persisted already. Use
+        :meth:`eth.abc.ChainDatabaseAPI.persist_unexecuted_block` to persist blocks that were not
+        executed.
         """
         ...
 
+    @abstractmethod
+    def persist_unexecuted_block(self,
```

Comment:
I don't necessarily think it's better but an alternate name that came to mind: `persist_historical_block`

Reply:
I like that the current name is in-your-face about not validating state transitions.

Reply:
Another idea would be `persist_nonexecuted_block` but I'm not sure if that is any better.